cirq-core/cirq/transformers/align.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transformer passes which align operations to the left or right of the circuit."""
+
+from typing import TYPE_CHECKING
+from cirq import circuits, ops
+from cirq.transformers import transformer_api
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@transformer_api.transformer
+def align_left(
+    circuit: 'cirq.AbstractCircuit',
+    context: 'cirq.TransformerContext' = transformer_api.TransformerContext(),
```

Comment:
Nit: unsafe default.

Reply:
What do you mean by unsafe default?

Reply:
Having an object that might be mutable as a default argument might allow mutations to that object to persist over time which we don't want. https://google.github.io/styleguide/pyguide.html#212-default-argument-values

Reply:
Made `context` an optional parameter with default value as None and explicit upgrade to default `TransformerContext()`.